Makefile

```diff
@@ -16,52 +22,70 @@ else
 PACK_VER:=$(VERSION)
 endif
 
-BINDIR=$(PWD)/install
+BINDIR=$(PWD)/install-$(ENGINE)
+
+BACKENDS_PATH ?= $(BINDIR)/backends
+
+#----------------------------------------------------------------------------------------------
+
+setup:
+	@echo Setting up system...
+	$(SHOW)./deps/readies/bin/getpy
+	$(SHOW)./system-setup.py
+
+.PHONY: all setup build clean deps pack pack_ramp pack_deps test
+
+all: build
 
-.PHONY: all clean deps pack rlec_runpath_fix
+ifeq ($(DEBUG),1)
+CMAKE_FLAGS += -DCMAKE_BUILD_TYPE=Debug
+endif
 
-all:
-ifeq ($(wildcard build/.),)
+build:
+ifeq ($(wildcard build/Makefile),)
 	mkdir -p build
 	cd build; \
-	cmake -DDEPS_PATH=../deps/install ..
+	cmake -DENGINE=$(ENGINE) -DDEPS_PATH=../deps/install-$(ENGINE) $(CMAKE_FLAGS) ..
 endif
-	$(MAKE) -C build && $(MAKE) -C build install
+	$(MAKE) -C build
+	$(MAKE) -C build install
 
 clean:
 ifeq ($(ALL),1)
-	rm -rf build deps
+	rm -rf build install deps/dlpack deps/install-$(ENGINE) deps/*.tar.gz deps/*.zip deps/*.tgz
 else
 	$(MAKE) -C build clean
 endif
 
-deps:
+deps fetch:
 	@echo Fetching dependencies...
 	@./get_deps.sh $(DEPS_FLAGS)
 
-# in pack: create ramp/redisai.so with RUNPATH set to /opt/redislabs/lib for RLEC compliance
-rlec_runpath_fix: 
-	@echo Fixing RLEC RUNPATH...
-	@mkdir -p $(BINDIR)/ramp
-	@cp -f $(BINDIR)/redisai.so $(BINDIR)/ramp/
-	# @patchelf --set-rpath $(REDIS_ENT_LIB_PATH) $(BINDIR)/ramp/redisai.so
+pack: pack_ramp pack_deps
 
-pack: rlec_runpath_fix
+pack_ramp:
 	@[ ! -z `command -v redis-server` ] || { echo "Cannot find redis-server - aborting."; exit 1; }
-	@[ ! -e $(REDIS_ENT_LIB_PATH) ] || { echo "$(REDIS_ENT_LIB_PATH) exists - aborting."; exit 1; }
-ifeq ($(wildcard build/pyenv/.),)
-	@virtualenv build/pyenv ;\
-	. ./build/pyenv/bin/activate ;\
-	pip install git+https://github.com/RedisLabs/RAMP
-endif
 	@echo "Building RAMP file ..."
 	@set -e ;\
-	. ./build/pyenv/bin/activate ;\
-	ln -fs $(PWD)/deps/install/lib/ $(REDIS_ENT_LIB_PATH) ;\
-	ramp pack -m $(PWD)/ramp.yml -o "build/redisai.{os}-{architecture}.${PACK_VER}.zip" $(BINDIR)/ramp/redisai.so 2>&1 > /dev/null ;\
-	rm $(REDIS_ENT_LIB_PATH)
-	@echo Done.
-	@echo "Building dependencies file redisai-dependencies.${PACK_VER}.tgz ..."
-	@cd deps/install/lib; \
-	tar pczf ../../../build/redisai-dependencies.${PACK_VER}.tgz *.so*
-	@echo Done.
+	RAMPOUT=$$(mktemp /tmp/ramp.XXXXXX) ;\
+	LD_LIBRARY_PATH=$(PWD)/deps/install-$(ENGINE)/lib \
+	ramp pack -m $(PWD)/ramp.yml -o "build/redisai.{os}-{architecture}.{semantic_version}.zip" $(BINDIR)/redisai.so 2> /dev/null | grep '.zip' > $$RAMPOUT ;\
+	tail -1 $$RAMPOUT > $(BINDIR)/PACKAGE ;\
+	rm -f $RAMPOUT ;\
+	echo "Done."
+
+pack_deps: pack_ramp
+	@echo "Building dependencies file ..."
+	@set -e ;\
+	PACK_FNAME=$$(basename `cat $(BINDIR)/PACKAGE`) ;\
+	ARCHOSVER=$$(echo "$$PACK_FNAME" | sed -e "s/^redisai\.\([^.]*\..*\)\.zip/\1/") ;\
+	cd install-$(ENGINE) ;\
+	find backends -name "*.so*" | xargs tar pczf redisai-dependencies.$$ARCHOSVER-$(ENGINE).tgz ;\
```

Comment:
Is it correct that we are going to have a `redisai-dependencies...tgz` that is specific for the ENGINE?

Just to make sure: only backends (`redisai_torch.so`, etc) need an ENGINE specified, not `redisai.so`.

Does the above mean that we'll have two tgz, one in which all backends are `cpu` and one in which all backends are `gpu`?